app/actions/ControlActions.js

```diff
@@ -322,18 +313,29 @@ function hexToBytes(hex) {
   return bytes;
 }
 
-function importScriptAction(purchaseTickets) {
+function importScriptAction(votingAddress, cb) {
   return (dispatch, getState) => {
     const { walletService } = getState().grpc;
     const { importScriptRequest } = getState().control;
     walletService.importScript(importScriptRequest,
-        function(err, importScriptResponse) {
-          if (err) {
+      function(err, importScriptResponse) {
+        if (err) {
+          if (!votingAddress && !cb) {
             dispatch(importScriptError(err + ' Please try again'));
           } else {
-            dispatch(importScriptSuccess(importScriptResponse, purchaseTickets));
+            var error = err + '';
+            if (error.indexOf('master private key') !== -1) {
+              dispatch(() => cb(err + ' Please try again.'));
+            } else {
+              err = err + '. This probably means you are trying to use a stakepool account that is already associated with another wallet.  If you have previously used a voting account, please create a new account (with any name) and try again.  Otherwise, please set up a new stakepool account for this wallet.';
```

Comment:
I guess I understand it when thinking about it (you mean it does not have to be named voting)